docs/index.rst

```diff
@@ -262,6 +262,17 @@ Options:
     rather than the raw enum - useful if you want to serialise ``model.dict()`` later (default: ``False``)
 :fields: extra information on each field, currently just "alias" is allowed (default: ``None``)
 :validate_assignment: whether to perform validation on assignment to attributes or not (default: ``False``)
+:allow_population_by_alias: whether or not a field may be populated by its alias, rather than strictly the name given
+    by the model attribute; please be sure to read the warning below before enabling this (default: ``False``)
+
+.. warning::
+
+   Think twice before enabling ``allow_population_by_alias``! Enabling it could cause previously correct code to become
+   subtly incorrect. As an example, say you have a field named ``card_number`` with the alias ``cardNumber``. With
+   population by alias disabled (the default), trying to parse an object with only the key ``cardNumber`` will fail.
+   However, if you enable population by alias, this previously-invalid object will now populate the ``card_number``
+   field from ``cardNumber``. This may be desired for some use cases, but in others (like the one given here, perhaps),
+   relaxing strictness could introduce bugs.
```

Comment:
wait this is the wrong way around I think, shouldn't it be

```
As an example, say you have a field named ``card_number`` with the alias ``cardNumber``. With
``allow_population_by_alias = False`` disabled, trying to parse an object with only the key 
``card_number`` will fail (the ``cardNumber`` field is required). However, if you enable population 
by alias, this previously-invalid object will now populate the ``card_number`` field from 
``card_number``. This may be desired or acceptable for some use cases, but in others it could 
introduce bugs.
```

?

Reply:
Oh my...thank you for catching that! 😳